ntoskrnl/config/cmkcbncb.c

```diff
@@ -1134,6 +1134,294 @@ DelistKeyBodyFromKCB(IN PCM_KEY_BODY KeyBody,
     if (!LockHeld) CmpReleaseKcbLock(KeyBody->KeyControlBlock);
 }
 
+/**
+ * @brief
+ * Unlocks a number of KCBs provided by a KCB array.
+ *
+ * @param[in] KcbArray
+ * A pointer to an array of KCBs to be unlocked.
+ */
+VOID
+NTAPI
+CmpUnLockKcbArray(
+    _In_ PULONG KcbArray)
+{
+    ULONG i, KcbCount;
+
+    /* Release the locked KCBs */
+    KcbCount = KcbArray[0];
+    for (i = KcbCount; i > 0; i--)
+    {
+        CmpReleaseKcbLockByKey(KcbArray[i]);
+    }
+}
+
+/**
+ * @brief
+ * Locks a given number of KCBs.
+ *
+ * @param[in] KcbArray
+ * A pointer to an array of KCBs to be locked.
+ * The count of KCBs to be locked is defined by the
+ * first element in the array.
+ *
+ * @param[in] KcbLockFlags
+ * Define a lock flag to lock the KCBs. CMP_LOCK_KCB_ARRAY_EXCLUSIVE
+ * indicates the KCBs are locked exclusively and owned by the calling
+ * thread, CMP_LOCK_KCB_ARRAY_SHARED indicates the KCBs are locked
+ * in shared mode by the owning threads.
+ */
+VOID
+NTAPI
+CmpLockKcbArray(
+    _In_ PULONG KcbArray,
+    _In_ ULONG KcbLockFlags)
+{
+    ULONG i;
+
+    /* Lock the KCBs */
+    for (i = 0; i < KcbArray[0]; i++)
```

Comment:
What about
```suggestion
    for (i = 1; i <= KcbArray[0]; i++)
```
and just using `KcbArray[i]` in the calls below?